Nightly- Fixes some errors

eloycoto opened the pull request:
Hi, 

Fixed few errors that has been fired in the last Nightly PR: 

- Issues on retrieve pods (Due the high load, it's a Kubernetes resource, no Cilium related)
- Issues with policy templates. I have seen this two times this week. 
- Added logs in `kub.CiliumPolicyRevision` that it was fired one time and I'm not sure why. With this change we can see the log output instead `jq` output.

eloycoto committed:
Nightly: Increase time to wait for Pods in Nightly.

I saw this problem in Nightly due the high load of servers with nearly
100 endpoints.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

eloycoto committed:
Nightly: Fix issues with policy import

Time to time, cilium policy import complains about the key of policies, not
always.If you use lowercase in the policy it works perfect:

Example failed policy:

```
 {
  "apiVersion": "cilium.io/v2",
  "kind": "CiliumNetworkPolicy",
  "metadata": {
    "name": "700e0976",
    "test": "L3:Ingress Label L4:Ingress Port 80 No protocol L7:Ingress policy /private/ Destination:Direct"
  },
  "specs": [
    {
      "EndpointSelector": {
        "matchlabels": { "id": "700e0976-afd3a30c" }
      },
      "Ingress": [
        {
          "FromEndpoints": [
            {
              "matchLabels": { "id": "700e0976-6cb50b02" }
            }
          ],
          "ToPorts": [
            {
              "Ports": [ { "port": "80" } ],
              "Rules": {
                "http": [
                  {
                    "method": "GET",
                    "path": "/private"
                  }
                ]
              }
            }
          ]
        }
      ],
      "Egress": null
    }
  ]
}
```

Error message:
```
vagrant@k8s1:~$ kubectl apply -f /home/vagrant/go/src/github.com/cilium/cilium/test/700e0976_policy.json                                                   The CiliumNetworkPolicy "700e0976" is invalid: []: Invalid value: map[string]interface {}{"apiVersion":"cilium.io/v2", "kind":"CiliumNetworkPolicy", "metadata":map[string]interface {}{"annotations":map[string]interface {}{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"cilium.io/v2\",\"kind\":\"
CiliumNetworkPolicy\",\"metadata\":{\"annotations\":{},\"name\":\"700e0976\",\"namespace\":\"default\",\"test\":\"L3:Ingress Label L4:Ingress Port 80 No protocol L7:Ingress policy /private/ Destination:Direct\"},\"specs\":[{\"Egress\":null,\"EndpointSelector\":{\"matchlabels\":{\"id\":\"700e0976-afd3a30c\"}},\"Ingres
s\":[{\"FromEndpoints\":[{\"matchLabels\":{\"id\":\"700e0976-6cb50b02\"}}],\"ToPorts\":[{\"Ports\":[{\"port\":\"80\"}],\"Rules\":{\"http\":[{\"method\":\"GET\",\"path\":\"/private\"}]}}]}]}]}\n"}, "name":"700e0976", "namespace":"default", "test":"L3:Ingress Label L4:Ingress Port 80 No protocol L7:Ingress policy /priv
ate/ Destination:Direct", "creationTimestamp":"2018-03-09T11:33:39Z", "uid":"b64c8b35-238d-11e8-8f5d-08002795cd5f", "selfLink":"", "clusterName":""}, "specs":[]interface {}{map[string]interface {}{"Egress":interface {}(nil), "EndpointSelector":map[string]interface {}{"matchlabels":map[string]interface {}{"id":"700e09
76-afd3a30c"}}, "Ingress":[]interface {}{map[string]interface {}{"FromEndpoints":[]interface {}{map[string]interface {}{"matchLabels":map[string]interface {}{"id":"700e0976-6cb50b02"}}}, "ToPorts":[]interface {}{map[string]interface {}{"Ports":[]interface {}{map[string]interface {}{"port":"80"}}, "Rules":map[string]i
nterface {}{"http":[]interface {}{map[string]interface {}{"method":"GET", "path":"/private"}}}}}}}}}}: validation failure list:
specs.endpointSelector in body is required
```

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

eloycoto committed:
Nightly: refactor kubectl.CiliumPolicyRevision to get logs

In Nightly, with high-load we had the following error message:

```
root/jenkins/workspace/Cilium-Master-Nightly-Tests-All/src/github.com/cilium/cilium/test/k8sT/PoliciesNightly.go:70
cannot apply network policy for ee9be4b2
Expected
    <*errors.errorString | 0xc42052e150>: {
        s: "Network policy cannot be imported prefix=ee9be4b2: Cannot get the revision command terminated with exit code 126\nparse error: Invalid numeric literal at line 1, column 4\n",
    }
to be nil
/root/jenkins/workspace/Cilium-Master-Nightly-Tests-All/src/github.com/cilium/cilium/test/helpers/policygen/models.go:341
```
From:
https://jenkins.cilium.io/job/Cilium-Master-Nightly-Tests-All/133/testReport/k8s-1/8/NightlyPolicies_PolicyEnforcement_default_L3_Ingress_Label_Invalid_L4_Egress_Port_80_TCP_L7_No_Policy_Destination_Direct/

Due the `jq` usage, the information that cilium send to us is not
available, with this change the output of `cilium policy get` will be
saved on logs, and help us in future debugging.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

aanm commented on test/helpers/policygen/models.go:
```diff
@@ -477,9 +480,9 @@ func (t *TestSpec) CreateCiliumNetworkPolicy() (string, error) {
 	}
 
 	type Spec struct {
-		EndpointSelector endpointSelector
```
Why not use `pkg/policy/api/rule.go:Rule` directly instead of `Spec`?

eloycoto replied:
done.

eloycoto committed:
Nightly: Using api.Rule instead custom Spec struct

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

eloycoto committed:
Nightly: clean unused struct

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

eloycoto committed:
Nightly: Fix typos

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

eloycoto committed:
Vagrantfile: Upload base image in test to version V37

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

eloycoto committed:
Nightly: Delete all default pods on NightlyExamples

Deleted all default pods created by PolicyGen, that are not deleted by
the end.

This code will disappear when AfterAll get merged.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>
